<?xml version="1.0"?>
<launch>
  <!-- start environment modelling -->
  <node pkg="nodelet" type="nodelet" name="pcl_manager"  args="manager" output="screen"/>
  

  
    <!-- Run a passthrough filter to clean NaNs -->
  <node pkg="nodelet" type="nodelet" name="passthrough" args="load pcl/PassThrough pcl_manager" output="screen">
    <remap from="~input" to="/kinect_rear2" />
    <rosparam>
      filter_field_name: z
      filter_limit_min: -10
      filter_limit_max: 10
      filter_limit_negative: False
    </rosparam>
  </node>
  
  <node pkg="cob_env_model" type="field_of_view_node" name="field_of_view_node" output="screen"/>

  <node pkg="nodelet" type="nodelet" name="aggregate_point_map" args="load cob_env_model/AggregatePointMap pcl_manager" output="screen">
  	<remap from="point_cloud2" to="/passthrough/output"/>
  	<remap from="point_cloud2_map" to="/aggregate_point_map/map"/>
  	<remap from="point_cloud2_aligned" to="/aggregate_point_map/pc_aligned"/>
  </node>  
  
  <!-- Run a VoxelGrid filter to clean NaNs and downsample the data -->
  <node pkg="nodelet" type="nodelet" name="voxel_grid_map" args="load pcl/VoxelGrid pcl_manager" output="screen">
    <remap from="~input" to="/aggregate_point_map/map" />
    <rosparam>
      filter_field_name: z
      filter_limit_min: -0.5
      filter_limit_max: 5
      filter_limit_negative: False
      leaf_size: 0.03
    </rosparam>
  </node>
  
</launch>

